spec/predicates/baseline.md

```diff
@@ -0,0 +1,367 @@
+as well as automated measurements of controls.
+
+## Model
+
+Baseline metadata is intended to capture the status of an open source software
+project relative to certain expected maturity levels. As open source software
+projects take many forms with many different types of assets, Baseline aims to
+flexibly define the sets of resources which may encompass a particular project,
+including:
+
+- Version controlled source code repositories (generally recorded _without_
+  reference to a specific commit)
+- Build and release pipelines or environments
+- Specific released software assets, such as binaries, libraries, or container
+  images
+- Accounts in distribution platforms such as NPM, DockerHub, or distributed
+  through websites
+- Documentation and governance materials for the project
+- Subprojects which contain one or more of the above and are managed through the
+  same group of collaborators.
+
+As Baseline aims to establish standard levels of project controls which can be
+applied by projects regardless of implementation, we expect that the most common
+mechanism for reporting controls compliance will be through a combination of
+automated and manual reporting. It is expected that the [level 1 criteria] will
+generally be amenable to automation on common hosting platforms, but that some
+criteria at the higher assessment levels may require manual attestation.
+
+In order to support clear attribution of controls measurement to actors, we
+expect that human and automated tooling will generate separate (possibly
+overlapping) baseline attestations, each mapped to a specific `author` and set
+of in-toto subjects. Consumers will consume an [in-toto attestation bundle]
+which contains all the attestations related to the project, and will be
+responsible for unifying the control claims across the attestations.
+
+## Schema
+
+The core metadata in the baseline predicate is the project assessment. A project
+assessment includes both a set of control assessments (a list with unique
+records for each control) and a reference to the entity which performed the
+assessment.
+
+```jsonc
+{
+  // Standard attestation fields:
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [{ "uri": "https://github.com/in-toto/attestation.git" }],
+
+  // Predicate:
+  "predicateType": "https://baseline.openssf.org/attestation/0.1",
+  "predicate": {
+    "author": { "uri": "https://github.com/evankanderson" },
+    "framework": "https://baseline.openssf.org/versions/2025-10-10",
+    "assessedAt": "2025-11-20T23:15:00Z",
+    "controls": [
+      {
+        "control": "OSPS-BR-03.01",
+        "result": "passed",
+        "evidence": [
+          {
+            "name": "https://github.com/in-toto/attestation",
+            "result": "passed",
+            "description": "HTTPS URL"
+          },
+          {
+            "name": "https://in-toto.io/",
+            "result": "passed",
+            "description": "HTTPS URL"
+          }
```

Comment:
Suggestion: Make the evidence entry a standard ResourceDescriptor, either direct or in a subtfield. This  makes it easier to shuffle around in the attestation handling systems as it is a data structure we already know how to handle and is expressive enough to capture many use cases :

```suggestion
          {
            "name": "Website URL",
            "uri": "https://in-toto.io/",
            "annotations": {
                "description": "HTTPS URL"
            }
          }
```

Reply:
The problem with using a ResourceDescriptor is that capturing text annotations is awkward (as `contents` is of type `[]byte`, which means base-64 encoding a human readable string in JSON, rendering it unreadable).

It looks like you're proposing using a well-known annotation to capture this information rather than storing it in `contents`?